src/python/dxpy/scripts/dx.py

```diff
@@ -3444,6 +3445,69 @@ def upgrade(args):
     except:
         err_exit()
 
+def generate_batch_inputs(args):
+    def eprint(*args, **kwargs):
+        print(*args, file=sys.stderr, **kwargs)
+
+    # Internally restricted maximum batch size for a TSV
+    MAX_BATCH_SIZE = 500
+    project, folder, _none = try_call(resolve_path, args.path, expected='folder')
+
+    # Parse input values
+    input_dict = dict([parse_input_keyval(keyeqval) for keyeqval in args.input])
+
+    # Call API for batch expansion
+    try:
+       api_result = dxpy.api.system_generate_batch_inputs({"project":  project, "folder": folder, "inputs": input_dict})['batchInputs']
+    except:
+       err_exit()
+
+    def chunks(l, n):
+        for i in range(0, len(l), n):
+            yield l[i:i + n]
+
+    successful = [b for b in api_result if 'error' not in b]
+    errors = [b for b in api_result if 'error' in b]
+
+    batches = list(chunks(successful, MAX_BATCH_SIZE))
+
+    eprint("Found {num_success} valid batch IDs matching desired pattern.".format(num_success=len(successful)))
+
+    input_names = input_dict.keys()
+
+    # Output TSV Batch
+    for i,batch in enumerate(batches):
+        def flatten_batch(b):
+            return [b['batchPattern']] + [ival['name'] for iname, ival in sorted(b['inputs'].items())] + [ival['ids'][0] for iname, ival in sorted(b['inputs'].items())]  
+
+        batch_fname = "{}.{:04d}.tsv".format(args.output_prefix, i)
+        with open(batch_fname, 'wb') as csvfile:
+            batchwriter = csv.writer(csvfile, delimiter='\t'.encode('ascii'))
```

Comment:
What kind of encoding are we supposed to use here? I used utf-8. I am now not sure which is correct.

Reply:
So if I recall, If I didn't encode this delimiter as ascii, the `csv.writer` errors out.  I'm assuming that since we import `unicode_literals` from `__future__` the '\t' string is UTF-8 which causes the writer to error out and that it requires an ASCII character delimiter.